### PR TITLE
A few warnings fixed

### PR DESCRIPTION
**Pull request by johanandren**
Compile warnings for streams and then a bunch when starting sbt. Not urgent in any way to get this merged.

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala [hunk 1]
```diff
@@ -82,7 +82,7 @@ class GraphStageLogicSpec extends StreamSpec with GraphInterpreterSpecKit {
     override def toString = "GraphStageLogicSpec.emitEmptyIterable"
   }
 
-  final case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
+  private case class ReadNEmitN(n: Int) extends GraphStage[FlowShape[Int, Int]] {
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: I think I saw something more similar to this, do `clean` `compile` in `akka-stream`

Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: The akka-stream one is tougher, it's the inner classes in `BatchingActorInputBoundary` which all access their outer scope. There unfinal them does not work and they cannot be private. One of the rabbit holes I explored today. Didn't figure out how I could solve that without changing more of it than I felt reasonable.

Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: ok, let's leave that for now

Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala]: Yes, those are trickier. Would be nice if we could just suppress those warnings.

**Pull request merged**